app/models/peer_review.rb

```diff
@@ -0,0 +1,28 @@
+class PeerReview < ActiveRecord::Base
+  belongs_to :reviewer, class_name: "Grouping", foreign_key: "reviewer_id"
+  belongs_to :reviewee, class_name: "Grouping", foreign_key: "reviewee_id"
+  belongs_to :result
+
+  validates_associated :reviewer
+  validates_associated :reviewee
+  validates_associated :result
+  validates_presence_of :reviewer
+  validates_presence_of :reviewee
+  validates_presence_of :result
+  validates_numericality_of :reviewer_id, only_integer: true,  greater_than: 0
+  validates_numericality_of :reviewee_id, only_integer: true,  greater_than: 0
+  validates_numericality_of :result_id, only_integer: true,  greater_than: 0
+  validate :prevent_reviewer_from_being_reviewee, :do_not_allow_one_nil_review
+
+  def do_not_allow_one_nil_review
+    if reviewer.nil? ^ reviewee.nil?
```

Comment:
I'm not sure what the purpose of this check is (seems like the presence is already validated).